### ISSUE TITLE
missing migration in pypi release

### ISSUE THREAD

**Issue by rsichny** [celery/django-celery-beat#115]
1. create clean django2.0 project
2. `pip install django-celery-beat`
3. add django_celery_beat to installed_apps
4. run `python manage.py migrate`

**Actual result:**
the warning appears:
```
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

**Expected:**
no warnings

The reason is missing migration in pypi release

**Comment by rsichny** [comment]
@DEKHTIARJonathan it works for development, but i don't feel like changing the code of library in the process of prod deployment...

**Comment by DEKHTIARJonathan** [comment]
It doesn't feel abnormal for me that you have to perform "makemigrations" in the prod deployment.
To be honest, I don't really if there is any best practice (well argumented) that in favor or against this practice.

**Comment by BITSOLVER** [comment]
The problem comes if you run "makemigrations" in production and the same migration appears in an update later on.  This will cause  "migrate" to fail due to a conflict, if the migration is identical to the one already applied then you could get away with running "--fake" on the updated version.

Personally I think the correct migrations should come from the library and you should not need to run "makemigrations" when using a 3rd party library.  A minor update would be really helpful.

**Comment by DEKHTIARJonathan** [comment]
It seems your explanation to be coherent up to me.
I then agree that the migrations should be included in the package.

**Comment by ilhnctn** [comment]
Just worked on my project too (Python 3.6, Django==1.11.9, djangorestframework==3.7.7, pip 9.0.1 from /usr/local/lib/python3.6/site-packages)
As @DEKHTIARJonathan stated, 
```sh
python manage.py makemigrations
python manage.py migrate
```

**Comment by rsichny** [comment]
> Personally I think the correct migrations should come from the library and you should not need to run "makemigrations" when using a 3rd party library. A minor update would be really helpful.

This is what i'm talking about

**Comment by mbaechtold** [comment]
This issue probably is a duplicate of https://github.com/celery/django-celery-beat/issues/115.

**Comment by pwmarcz** [comment]
Seconding the request for a release. 1.1.0 is broken (not only a migration file is missing, but it's impossible to provide a correct one yourself due to https://github.com/celery/django-celery-beat/commit/5bccc2dd9f5dd30f19cff30ffcf449c5fca91897) and blocks us from upgrading to Django 2.0.

**Comment by muhtarc4n** [comment]
Can't we just have a bug release (v1.1.1) for this issue. We enabled migration tests and now they are all failing because of missing migration. We can't just make migration and create migration file in dependency folder. We don't have permission which is very good practice.

**Comment by DEKHTIARJonathan** [comment]
I support this @thedrow

**Comment by muhtarcanrb** [comment]
And I think fix is already in master branch. I can see 5th migration in migrations folder.

**Comment by muhtarc4n** [comment]
And I think fix is already in master branch. I can see 5th migration in migrations folder.

**Comment by mbaechtold** [comment]
Until there is a new release, you can still `pip install` the package directly from GitHub. You can even specify a specific commit hash while doing so, no need to install from the master branch:

```
pip install git+https://github.com/celery/django-celery-beat@ba48402770eead01f059cb364fd522996761d003#egg=django-celery-beat
```

**Comment by DEKHTIARJonathan** [comment]
Come on, everything would be so much easier if a minor release were released. 
Additionnally the existing version is not working properly: #113 #117

**Comment by muhtarc4n** [comment]
@mbaechtold it is impossible to install this dependency this way in our company. We put dependency in requirements file and it pulls version from pypi repository. 

There is a bug and project needs bug fix version.

**Comment by mbaechtold** [comment]
@muhtarc4n, you can install packages from GitHub by specifying them in your `requirements.txt` too. See the docs of `pip`, especially the part about [Requirements Files](https://pip.pypa.io/en/stable/user_guide/#requirements-files):

> For example, suppose a dependency, SomeDependency from PyPI has a bug, and you can't wait for an upstream fix. [..]

You might also have a look at the supported forms when [installing from Git](https://pip.pypa.io/en/stable/reference/pip_install/?highlight=vcs#git).

**Comment by DEKHTIARJonathan** [comment]
I guess what @muhtarc4n say, is that it is **not authorized** to do such a thing in his company.
Which is, by the way, my situation. Only "official" versions available through PyPI are authorized.

**Comment by Ivo-Donchev** [comment]
This was caused by a `Solar.event` CharField with choices get from an unordered set which causes the new migration every time you run `makemigrations`. (same choices but different order every time)

It's seems that it's already fixed here: 
`https://github.com/celery/django-celery-beat/pull/92/files`

**Comment by DEKHTIARJonathan** [comment]
Absolutely, but that's a **very good** reason to releazse a small patch that fixes the problem.

**Comment by Ivo-Donchev** [comment]
@mbaechtold Is there a chance for a bugfix release soon :) ?

**Comment by mbaechtold** [comment]
@Ivo-Donchev, I'm sorry, I do not have the permission for releasing this Python package.

**Comment by Ivo-Donchev** [comment]
:disappointed: @mbaechtold  I suppose specifying requirements' version of this package to the lastest git commit in master is a possible solution then...

Ok, thanks anyway

**Comment by mbaechtold** [comment]
You could even go further and host your own PyPI server, fork this package and release it on your private PyPI server.

**Comment by Ivo-Donchev** [comment]
@mbaechtold It's always an option :slightly_smiling_face:  but is too much overkill to do it just to avoid an empty migration when running `makemigrations` :D

**Comment by KonnikPahoni** [comment]
Your models in app(s): 'django_celery_beat' have changes that are not yet reflected in a migration

![image](https://user-images.githubusercontent.com/13286998/189470042-36ac4e8b-ece5-4c6f-b8f5-a250ff089ede.png)

It is a really bad practice, appears in django-celery-beat==2.2.0

**Comment by auvipy** [comment]
did you check https://github.com/celery/django-celery-beat/releases/tag/v2.3.0

**Comment by KonnikPahoni** [comment]
> did you check https://github.com/celery/django-celery-beat/releases/tag/v2.3.0

I still use 2.2.0 because of this error on Windows:

![image](https://user-images.githubusercontent.com/13286998/189473524-87b174f2-b104-4446-8dec-8bac673c8a80.png)

When I change 2.2.0 to 2.3.0, I get:

```
...
File "/usr/local/lib/python3.10/site-packages/django_celery_beat/models.py", line 8, in <module>
drugstat.worker_1     |   File "/usr/local/lib/python3.10/zoneinfo/_common.py", line 28, in load_data
drugstat.worker_1     |     header = _TZifHeader.from_file(fobj)
drugstat.worker_1     |   File "/usr/local/lib/python3.10/zoneinfo/_common.py", line 146, in from_file
drugstat.worker_1     |     raise ValueError("Invalid TZif file: magic not found")
drugstat.worker_1     | ValueError: Invalid TZif file: magic not found
```

I use Dockerfile with python:3.10 on Windows (Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.31).

**Comment by auvipy** [comment]
you might have messed up with dependencies? we haven't got any report of related bugs

**Comment by KonnikPahoni** [comment]
> you might have messed up with dependencies? we haven't got any report of related bugs

No idea, spent several hours on it, was not able to use 2.3.0 on WSL-2.
